internal/database/repo_update_jobs.go

```diff
@@ -0,0 +1,180 @@
+package database
+
+import (
+	"context"
+	"time"
+
+	"github.com/keegancsmith/sqlf"
+	"github.com/lib/pq"
+
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/database/basestore"
+	"github.com/sourcegraph/sourcegraph/internal/database/dbutil"
+	"github.com/sourcegraph/sourcegraph/internal/executor"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+)
+
+type RepoUpdateJobStore interface {
+	Handle() *basestore.Store
+	Create(ctx context.Context, opts RepoUpdateJobOpts) (types.RepoUpdateJob, bool, error)
+	List(ctx context.Context, opts ListRepoUpdateJobOpts) ([]types.RepoUpdateJob, error)
+	SaveUpdateJobResults(ctx context.Context, jobID int, opts SaveUpdateJobResultsOpts) error
+}
+
+type repoUpdateJobStore struct {
+	db *basestore.Store
+}
+
+func RepoUpdateJobStoreWith(other basestore.ShareableStore) RepoUpdateJobStore {
+	return &repoUpdateJobStore{db: basestore.NewWithHandle(other.Handle())}
+}
+
+func (s *repoUpdateJobStore) Transact(ctx context.Context) (*repoUpdateJobStore, error) {
+	tx, err := s.db.Transact(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &repoUpdateJobStore{
+		db: tx,
+	}, nil
+}
+
+func (s *repoUpdateJobStore) Done(err error) error {
+	return s.db.Done(err)
+}
+
+func (s *repoUpdateJobStore) Handle() *basestore.Store {
+	return s.db
+}
+
+type RepoUpdateJobOpts struct {
```

Comment:
I think it's more common to use the verb in the name of the opts, right?

```suggestion
type CreateRepoUpdateJobOpts struct {
```